servers/quarkus-server/src/main/java/org/projectnessie/server/catalog/CatalogProducers.java

```diff
@@ -106,34 +109,36 @@ public void closeSdkHttpClient(@Disposes @CatalogS3Client SdkHttpClient client)
   @SuppressWarnings({"unchecked", "rawtypes", "UnnecessaryLocalVariable"})
   @Produces
   @Singleton
-  public S3Options<S3BucketOptions> s3Options(CatalogS3Config s3Config) {
+  public S3Options<S3BucketOptions, S3NamedBucketOptions> s3Options(CatalogS3Config s3Config) {
     S3Options opts = s3Config;
-    S3Options<S3BucketOptions> r = opts;
+    S3Options<S3BucketOptions, S3NamedBucketOptions> r = opts;
```

Comment:
Could we add two things here (and similarly for GCS + ADLS):

1. Rebuild S3Options with the names from the named-bucket-options
2. While doing that, validate that map-keys and bucket-names do not

This could remove the need to iterate over all buckets in S3Options.

WDYT?

Reply:
I like the idea and am trying to do as you suggested. However, the Immutables library is refusing to generate a method `Builder from(S3Options<S3BucketOptions, S3NamedBucketOptions>)` in `S3ProgrammaticOptions` – I think it's a limitation of the library because the parent interface is generic. So that method must be implemented manually, with the risk of us forgetting to copy some of the properties that could be added later. Wdyt?

Reply:
I _think_ there's a way to get a builder-instance from an immutable - just not enabled (or visible) by default.

Reply:
Also: the `from()` functions don't really "care" about the input type - as long as it is somehow compatible. But `S3ProgrammaticOptions` would do a "deep `from()`", so all maps need to be updated anyways.

Reply:
I have a solution, which involves removing the generic type parameters from `S3Options` and the like. Would you like me to push that change?

Reply:
Sure, please! Your code's a bit invisible over here ;) ;)